src/Mod/Part/App/TopoShapeExpansion.cpp

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/****************************************************************************
+ *                                                                          *
+ *   Copyright (c) 2022 Zheng, Lei <realthunder.dev@gmail.com>              *
+ *   Copyright (c) 2023 FreeCAD Project Association                         *
+ *                                                                          *
+ *   This file is part of FreeCAD.                                          *
+ *                                                                          *
+ *   FreeCAD is free software: you can redistribute it and/or modify it     *
+ *   under the terms of the GNU Lesser General Public License as            *
+ *   published by the Free Software Foundation, either version 2.1 of the   *
+ *   License, or (at your option) any later version.                        *
+ *                                                                          *
+ *   FreeCAD is distributed in the hope that it will be useful, but         *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU       *
+ *   Lesser General Public License for more details.                        *
+ *                                                                          *
+ *   You should have received a copy of the GNU Lesser General Public       *
+ *   License along with FreeCAD. If not, see                                *
+ *   <https://www.gnu.org/licenses/>.                                       *
+ *                                                                          *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+
+#include "TopoShape.h"
+#include "TopoShapeCache.h"
+
+FC_LOG_LEVEL_INIT("TopoShape",true,true)  // NOLINT
+
+namespace Part
+{
+
+void TopoShape::initCache(int reset) const{
+    if(reset>0 || !_cache || _cache->isTouched(_Shape)) {
```

Comment:
This is new file, it should be probably auto formatted.

Reply:
Yeah, I'm not sure why it wasn't -- probably because Part is excluded right now. I'll tell cLion to do it anyway.